File: 012/012.c
#include <stdio.h>
#define DIVISOR 500

int calculate_next_triangle(int);

int main()
{
   int divisor_count = 0;
   int natural_num = 1;
   int triangle_num;
   int i;
   while(divisor_count < DIVISOR)
   {
        divisor_count = 0;
        triangle_num = calculate_next_triangle(natural_num);
        for(i=1; i <= triangle_num/2; i++)
        {
            if(triangle_num % i == 0)
            {
                divisor_count ++;
            }
        }
        divisor_count++;
        natural_num++;
   }
   printf("Result: %d",triangle_num);
   return 0;
}

int calculate_next_triangle(int number)
{
    return (number*(number+1)) / 2;
}

File: 012/012.js
(function () {
    const DIVISOR = 500;
    var lastAddedNaturalNumber = 1;
    var triangleNumber = 0;
    var divisorCount = 0;

    while (divisorCount < DIVISOR) {
        divisorCount = 0;
        calculateNextTriangleNumber();
        for (var i = 1; i <= triangleNumber/2; i++) {
            if(triangleNumber % i === 0) {
                divisorCount++;
            }
        }
        divisorCount++;
    }
    console.log(triangleNumber);

    function calculateNextTriangleNumber() {
        triangleNumber =  triangleNumber + lastAddedNaturalNumber;
        lastAddedNaturalNumber++;
    }
}());